text/0000-lts-release-strategy.md

```diff
@@ -0,0 +1,90 @@
+
+If a major release is not delivered for an extended period of time, the maintainers may designate a subsequent minor release as the next LTS release, for example if v3.1 is a LTS release and there is no need to increment to 4.0 for several quarters, the maintainers may decide to designate v3.4 as the next LTS release.
+
+## LTS 3rd digit patch releases
+
+For LTS releases, 3rd digit patch releases will be provided for bug and security fixes at least every three months, but may be more frequent if important fixes become available sooner. Significant new features and other changes will not be included in 3rd digit patch releases, in order to ensure the stability of the LTS release and reduce the risk of functional regressions and bugs.
+
+It is worth noting that the v1.4 LTS release did not follow the above principles. Had the v1.4 LTS release followed the above principles, the Raft ordering service feature would likely have been introduced in a v1.5 release, and Kafka to Raft migration in a v1.6 release. We mention only to illustrate the above principles.
+
+## LTS release duration
+
+Once a *subsequent* LTS release is designated, users can expect 3rd digit patch releases to address critical bugs on the *prior* LTS release for approximately 9 more months. The maintainers will determine which fixes should to be backported to the latest LTS release, versus which fixes need to be further backported to the prior LTS release. The overlap period is intended to provide users a time window to upgrade their nodes, and any channel capabilities (if required).
+
+## LTS release diagram
+The following diagram represents the LTS release cadence and duration using hypothetical Fabric releases:
+![LTS release diagram](../images/lts_release_diagram.png)
+
+## LTS and channel capabilities
+
+Channel capabilities will remain available for two LTS releases, but may be removed in subsequent LTS releases.  For example, v1.x capabilities such as legacy chaincode will be available through v2.2 (assuming v2.2 is a LTS release), but are expected to be removed before the subsequent LTS release. This provides one full LTS cycle for channels to transition to new capabilities. For example, a production Fabric network may perform a rolling upgrade from v1.4.x nodes to v2.2.x nodes, then enable v2.x channel capabilities to take advantage of the new chaincode lifecycle, and then eventually upgrade to v3.x LTS nodes, where the legacy chaincode lifecycle may not be available.
+
+## LTS and upgrade testing
+
+The Fabric project expects to test single step upgrades (e.g. v2.0 to v2.1), and LTS to next-LTS upgrades (e.g. v1.4.x to v2.2). Other upgrade paths will not be tested and are not guaranteed to work.
```

Comment:
I understand that support for LTS-next-LTS upgrade is a must to have but we have TODOs for certain code removal in ledger specifically to support this. This is because the upgrade is more like a peer failure and recovery. As it affects ledger, we have some codes in ledger just to manage upgrade from one LTS to another. I think we need to live with it.

Reply:
@cendhu  I'm not understanding the main point here? What is the takeaway?

Reply:
This was just a comment I had on LTS-next-LTS rolling upgrade. 

Whenever we add features to the ledger code or remove code or refactor code, we always need to keep LTS-next-LTS in mind. This could introduce bugs as our LTS timeline is long and we might change the code a lot in between. Takeaway can be that we need to very extra cautious while changing the ledger code.

Reply:
Agreed!